python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_check_selection.py

```diff
@@ -0,0 +1,140 @@
+        "orders",
+        "raw_customers",
+        "raw_orders",
+        "raw_payments",
+        "stg_customers",
+        "stg_orders",
+        "stg_payments",
+    ]
+}
+
+
+ALL_CHECK_KEYS = {
+    AssetCheckKey(asset_key=AssetKey([asset_name]), name=check_name)
+    for asset_name, check_name in [
+        ("customers", "not_null_customers_customer_id"),
+        ("customers", "unique_customers_customer_id"),
+        (
+            "orders",
+            "accepted_values_orders_status__placed__shipped__completed__return_pending__returned",
+        ),
+        ("orders", "not_null_orders_amount"),
+        ("orders", "not_null_orders_bank_transfer_amount"),
+        ("orders", "not_null_orders_coupon_amount"),
+        ("orders", "not_null_orders_credit_card_amount"),
+        ("orders", "not_null_orders_customer_id"),
+        ("orders", "not_null_orders_gift_card_amount"),
+        ("orders", "not_null_orders_order_id"),
+        ("orders", "relationships_orders_customer_id__customer_id__ref_customers_"),
+        ("orders", "unique_orders_order_id"),
+        ("stg_customers", "not_null_stg_customers_customer_id"),
+        ("stg_customers", "unique_stg_customers_customer_id"),
+        (
+            "stg_orders",
+            "accepted_values_stg_orders_status__placed__shipped__completed__return_pending__returned",
+        ),
+        ("stg_orders", "not_null_stg_orders_order_id"),
+        ("stg_orders", "unique_stg_orders_order_id"),
+        (
+            "stg_payments",
+            "accepted_values_stg_payments_payment_method__credit_card__coupon__bank_transfer__gift_card",
+        ),
+        ("stg_payments", "not_null_stg_payments_payment_id"),
+        ("stg_payments", "unique_stg_payments_payment_id"),
+    ]
+}
+
+
+def test_all():
+    asset_selection = build_dbt_asset_selection([my_dbt_assets], dbt_select="fqn:*")
+    assert asset_selection.resolve(asset_graph) == ALL_ASSET_KEYS
+    assert asset_selection.resolve_checks(asset_graph) == ALL_CHECK_KEYS
+
+
+def test_tag():
+    asset_selection = build_dbt_asset_selection([my_dbt_assets], dbt_select="tag:data_quality")
+    assert asset_selection.resolve(asset_graph) == set()
+    assert asset_selection.resolve_checks(asset_graph) == {
+        AssetCheckKey(asset_key=AssetKey(["customers"]), name="unique_customers_customer_id")
+    }
+
+    asset_selection = build_dbt_asset_selection([my_dbt_assets], dbt_exclude="tag:data_quality")
+    assert asset_selection.resolve(asset_graph) == ALL_ASSET_KEYS
+    assert asset_selection.resolve_checks(asset_graph) == ALL_CHECK_KEYS - {
+        AssetCheckKey(asset_key=AssetKey(["customers"]), name="unique_customers_customer_id")
+    }
+
+
+def test_selection_customers():
+    asset_selection = build_dbt_asset_selection([my_dbt_assets], dbt_select="customers")
+    assert asset_selection.resolve(asset_graph) == {AssetKey(["customers"])}
+    # all tests that reference model customers- includes a relationship test on orders
```

Comment:
I'm a little surprised by this, am I interpreting correctly?

Reply:
[The default mode is `eager`](https://docs.getdbt.com/reference/node-selection/test-selection-examples?indirect-selection-mode=eager), so this matches that behavior. Perhaps we should augment `build_dbt_asset_selection` to include `indirect_selection` as a parameter?

Reply:
Yeah seems like a likely request in the future